dss/storage/validator.py

```diff
@@ -9,15 +13,73 @@
 import ssl
 from typing import List
 import logging
+from dss import Replica, Config
+from hashlib import sha1
+import requests
+
+
+DSS_Draft4Validator = validators.create(meta_schema=_utils.load_schema("draft4"),
+                                        validators={u"$ref": _validators.ref,
+                                                    u"additionalProperties": _validators.additionalProperties,
+                                                    u"properties": _validators.properties_draft4,
+                                                    u"required": _validators.required_draft4
+                                                    },
+                                        version="draft4"
+                                        )
+
+
+class S3UrlCache:
+    """Used by the resolver to retrieve the url contents from either a url or an s3 cloud bucket"""
```

Comment:
How do you know only a few MB can be stored?

Reply:
Storage in S3 is not a problem but the fact that you're reading the entire URL's content into memory might be. If you get a URL pointing to a 1G file you'll be using 1G of memory (or a multiple of that, e.g. when decoding the string from UTF-8). You'll also block for a minute or so downloading the file. Files of that size are more and more impractical to store in memory and are usually streamed rather than loaded into memory, especially in a resource constraint environment like a lambda.

Reply:
https://docs.python.org/3/library/shutil.html#shutil.copyfileobj

is a good way to limit the amount of data read from a file handle. You could use this to prevent the lambda from being DOSed by freakishly large URLs.